libs/initScripts/computeRSet_with_HCs.js

```diff
@@ -0,0 +1,346 @@
+    }
+  }
+
+  return { id, endpoint, status: 500 }
+}
+
+const writeDataToFile = (spIdToEndpointAndCount, userIdToRSet) => {
+  fs.writeFile('computeRSet_Data.txt', `\n<userIds - replica sets>:\n` + JSON.stringify(userIdToRSet), err => {
+    if (err) console.error(`Error with writing <userIds - replica sets> data:`, err)
+    else console.log('Saved <userIds - replica sets> data')
+  })
+  fs.appendFile('computeRSet_Data.txt', '\n<service providers - number of times selected as secondary>:\n' + JSON.stringify(spIdToEndpointAndCount), err => {
+    if (err) console.error(`Error with writing <service providers - number of times selected as secondary> data:`, err)
+    else console.log('Saved <service providers - number of times selected as secondary> data')
+  })
+}
+
+const syncSecondary = async ({ primary, secondary, wallet }) => {
+  try {
+    await axios({
+      baseURL: secondary,
+      url: '/sync',
+      method: 'post',
+      data: {
+        wallet: [wallet],
+        creator_node_endpoint: primary,
+        immediate: false /* whether or not this is a blocking request and handled right away */
+      }
+    })
+  } catch (e) {
+    console.error(`Could not sync from primary=${primary} to secondary=${secondary} for wallet=${wallet}`, e)
+  }
+}
+
+const getClockValue = async ({ id, endpoint, wallet, timeout = 5000 }) => {
+  try {
+    const clockValue = await CreatorNode.getClockValue(endpoint, wallet, timeout)
+    return { clockValue, id, endpoint, wallet }
+  } catch (e) {
+    console.error(`Could not fetch clock value for wallet=${wallet} from endpoint=${endpoint}`, e.message)
+    return { clockValue: null, id, endpoint, wallet }
+  }
+}
+
+const setReplicaSet = async ({
+  audiusLibs,
+  primary,
+  secondary1,
+  secondary2,
+  userId
+}) => {
+  // Update in new contract
+  let tx = await audiusLibs.contracts.UserReplicaSetManagerClient.updateReplicaSet(
+    userId,
+    primary.spId,
+    [secondary1.spId, secondary2.spId]
+  )
+
+  console.log('tx for updating rset', tx)
+  const newCreatorNodeEndpoint = CreatorNode.buildEndpoint(
+    primary.endpoint, [secondary1.endpoint, secondary2.endpoint]
+  )
+  await audiusLibs.User.waitForCreatorNodeEndpointIndexing(userId, newCreatorNodeEndpoint)
+}
+
+let start
+const run = async () => {
+  // Set up libs
+  start = Date.now()
+  const audiusLibs = await configureAndInitLibs()
+
+  audiusLibs.discoveryProvider.setEndpoint(DISCOVERY_NODE_ENDPOINT)
+
+  // Get all users that do not have a replica set assigned
+  const numOfUsers = (await axios({
+    method: 'get',
+    url: '/latest/user',
+    baseURL: DISCOVERY_NODE_ENDPOINT
+  })).data.data
+
+  console.log(`There are ${numOfUsers} users on Audius\n`)
+
+  // Batch DP /users calls
+  let userIdToWallet = {}
+  let i
+  for (i = 0; i <= numOfUsers; i = i + NUM_USERS_PER_BATCH_REQUEST) {
+    console.log(`Processing users batch range ${i} to ${i + NUM_USERS_PER_BATCH_REQUEST}...`)
```

Comment:
if we run this against production, will all 110k users be processed in batches? I suspect this script will OOM

Reply:
let's add a flag to optionally accept a min userId? or something similar because i don't think this will work against production

Reply:
and a max numUsers, basically some more fine grain control around the processing

Reply:
before doing the above, let's actually perform operations in batches - not sure why we'd want to split these two things apart, https://github.com/AudiusProject/audius-protocol/pull/1290/files#r592593128